riscv/src/compiler.rs

```diff
@@ -682,6 +696,18 @@ fn preamble() -> String {
         { Y_7bit } in { seven_bit };
     }
 
+    // Input is a 32 bit unsigned number. We check bit 15 and set all higher bits to that value.
+    instr sign_extend_16_bits Y -> X {
+        Y_15bit = X_b1 + Y_7bit * 0x100,
+
+        // wrap_bit is used as sign_bit here.
+        Y = Y_15bit + wrap_bit * 0x8000 + X_b3 * 0x10000 + X_b4 * 0x1000000,
+        X = Y_15bit + wrap_bit * 0xffff8000
+    }
+    constraints{
+        col witness Y_15bit;
```

Comment:
Theres a lookup in the other sign extend instr's constraints, why is this different?

Reply:
X_b1 is constrained, Y_7bit is constrained, so X_b1 + Y_7bit * 0x100 is also constrained, no need for a lookup here.